nightfall-deployer/contracts/Shield.sol

```diff
@@ -129,37 +124,37 @@ contract Shield is Stateful, Structures, Config, Key_Registry, ReentrancyGuardUp
     /**
   This function enables funds to be withdrawn, once a block is finalised
   @param b - the block containing the Withdraw transaction
-  @param ts - array of the transactions contained in the block
+  @param t - array of the transactions contained in the block
   @param index - the index of the transaction that locates it in the array of Transactions in Block b
   TODO do we need to pass in  all the block data?
   */
 
     function finaliseWithdrawal(
         Block calldata b,
-        uint256 blockNumberL2,
-        Transaction[] calldata ts,
-        uint256 index
+        bytes32 transactionsHash,
+        Transaction calldata t,
+        uint256 index,
+        bytes32[6] memory siblingPath
```

Comment:
Is fixing the length of the `siblingPath` problematic if we have blocks of different size? I guess this should be dynamic based on the number of transactions in a block.